src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -254,23 +253,26 @@ public String getQuestionResultStatisticsHtml(
 
         Map<String, List<Integer>> recipientRanks = generateOptionRanksMapping(responses);
 
+        Map<String, Integer> recipientOverallRank = generateNormalizedOverallRankMapping(recipientRanks);
+
         Map<String, List<Integer>> recipientRanksExcludingSelf = getRecipientRanksExcludingSelf(responses);
+
+        Map<String, Integer> recipientOverallRankExceptSelf =
+                generateNormalizedOverallRankMapping(recipientRanksExcludingSelf);
+
         Map<String, Integer> recipientSelfRanks = generateSelfRankForEachRecipient(responses);
 
         String fragmentTemplateToUse = FormTemplates.RANK_RESULT_STATS_RECIPIENTFRAGMENT;
         String templateToUse = FormTemplates.RANK_RESULT_RECIPIENT_STATS;
 
-        DecimalFormat df = new DecimalFormat("#.##");
-
         recipientRanks.forEach((participantIdentifier, ranks) -> {
 
-            double average = computeAverage(ranks);
             String ranksReceived = getListOfRanksReceivedAsString(ranks);
-
+            String overallRank = Integer.toString(recipientOverallRank.get(participantIdentifier));
             String name = bundle.getNameForEmail(participantIdentifier);
             String teamName = bundle.getTeamNameForEmail(participantIdentifier);
-            String userAverageExcludingSelfText =
-                    getAverageExcludingSelfText(df, recipientRanksExcludingSelf, participantIdentifier);
+            String overallRankExceptSelf = recipientOverallRankExceptSelf.containsKey(participantIdentifier)
+                    ? Integer.toString(recipientOverallRankExceptSelf.get(participantIdentifier)) : "-";
```

Comment:
Do we have test cases covered for this condition - Overall rank return `-` instead of a rank? Similar question for other place in your PR

Reply:
Thx for the review! This case, namely the return of `-` in absence of data for `Overall Rank Except Self` is covered for a rank recipient question test in `FeedbackSessionsLogicTest`. The absence of data for `Overall Rank` of an option/ recipient will result in the absence of this option/recipient in the statistic summary shown, and I just edited one of the two test cases for `Rank Question` to include this scenario.

Reply:
I see. That means that `-` will never be shown right?

Reply:
Emmmmm `-` for `Overall Rank Except Self` is shown when there is no other people ranking this recipient excluding himself. 
![image](https://user-images.githubusercontent.com/28526038/36128834-60daf79c-109f-11e8-88c0-1f4caaf86530.png)

Reply:
@wenmogu Then can u add that test cases?

Reply:
oh the[ old test case at `FeedbackSessionLogicTest` line 1845](https://github.com/TEAMMATES/teammates/pull/8400/files#diff-9a00a8a9d3c36abb0f74c76ffe4946f3L1845) actually suffices :D

Reply:
You are correct :)